R/transformAbundance.R

```diff
@@ -0,0 +1,400 @@
+#' Transform Abundance
+#'
+#' This function applies transformation to abundance table, which is returned.
+#' Additionally, the information can be directly stored in the \code{assay}.
+#'
+#' @param x
+#' A \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' object.
+#'
+#' @param abund_values
+#' A single character value for selecting the
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}}
+#'   to be transformed.
+#'
+#' @param transform
+#' A single character value for selecting the transformation method.
+#'
+#' @param name
+#' A single character value specifying the name of transformed abundance table.
+#'
+#' @param pseudocount
+#' Boolean or numeric value deciding whether pseudocount is added. Numerical
+#' value specifies the value of pseudocount. When selecting pseudocount==TRUE,
+#' pseudocount is 1.
+#'
+#' @param scale
+#' A numeric value for scaling. Values are multiplied with specified value.
+#'
+#' @param target
+#' A single character value specifying, if z transformation is applied
+#' for samples or features.
+#'
+#' @details
+#' \code{transformAbundance} applies transformation to abundance table.
+#' Transformation methods include
+#'
+#' \itemize{
+#'
+#'     \item{"relabundance" }{Transforms abundances to relative.
+#'     ($\frac{x}{x_{tot}}$, where $x$ is a single value and $x_{tot}$ is the sum of
+#'     all values.)}
```

Comment:
I think this whole thing whould be merge with the `relAbundanceCounts` function. The parallels are already there and and additional man page is probably not necessary.

Reply:
Hi, now the code is quite ready. I think there is only one bigger thing left, and it is this. What do you mean by merging? When user chooses method = "relabundance", should I check that and then just get the table with relAundanceCounts()? 

Like this:

```
if(method=="relabundance"){
    relAbundanceCounts(x, abund_values, name)
}else{
    # Get transformed table
     transformed_table <- .get_transformed_table(assay = assay(x, abund_values),
                                                          method = method,
                                                          pseudocount = pseudocount,
                                                          scalingFactor = scalingFactor)

      # Assign transformed table to assays
      assay(x, name) <- transformed_table

        return(x)
}
```

Reply:
Yes by merging we mean that it should be accessible like this. 

I would not export the relative abundance methods separately (unless we export all specific transformations in the same way but I do not see a clear need).

Reply:
I mean with merging, that all the functionality of the `relAbundanceCounts` function and file, should be merge with `transformAbundance`. There should only, one file, one man page and one implementation. You can export a few different methods, but the results should only be generated via one pipeline.

So in order to merge it, copy paste the stuff you need into your file. Make sure, that the man page contains all the details from `relAbundanceCounts`.

I also want to point out, that there is the open conversation whether `transformCounts` wouldn't be a better name for the function.

Reply:
Yes, I agree. Felix put it more exactly.

I have in fact commented in the open conversation that I support `transformCounts`.

Reply:
Just to make maybe more descriptive: `transformCounts` should be the main function. An implementation for `relAbundanceCounts` could look like this:

```
setMethod("relAbundanceCounts", signature = c(x = "SummarizedExperiment"),
function(x, ...){
  transformCounts(x, method="relabundance", ...)
})
```

Reply:
Do we really need a separate `relAbundanceCounts` wrapper (and other similar ones) or would just `transformCounts` be enough?

Reply:
I would like to keep. It is probably the most used transform functions. In the end it will be 6 lines of code.

Reply:
ok (still thinking if `relAbundanceCounts` is the most descriptive name for this but we can keep it unless anyone comes up with other suggestions)

Reply:
I would also agree with @FelixErnst . Maybe a bit controversial, but do you see the possibility of using these options `relProportion` or `proportionalCounts`?

Reply:
relAbundanceCounts merged to transformCounts